dascore/io/dasdae/core.py

```
@@ -73,10 +74,16 @@ def write(
         with contextlib.suppress(ValueError):
             resource.create_group("waveforms")
         waveforms = resource["waveforms"]
-        # write new patches to file
+        # write new patches to file, ensuring unique group names within this
+        # batch so same-named patches (e.g. gap-split siblings that differ
+        # only along a non-named dimension) don't overwrite each other.
         patch_names = get_patch_names(patches).values
+        counts: dict[str, int] = {}
         for patch, name in zip(patches, patch_names):
```

Ruff (0.15.20)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)